tests/ci/azure_pipeline_test/dsvm_linux_template.yml

```diff
@@ -21,51 +21,103 @@ parameters:
 - name: pytest_params
   type: string
   default: ""
+- name: install
+  type: string
+  default: ""
+- name: package
+  type: string
+  default: ""
 
 jobs:
-- job: test
+- job: "${{ parameters.conda_env }}" # Job name can't have spaces
   displayName: "${{ parameters.task_name }}"
   timeoutInMinutes: ${{ parameters.timeout }}
   pool:
     name: $(Agent_Pool)
 
   steps:
+
   - script: |
       echo "##vso[task.prependpath]/data/anaconda/bin"
-      conda create -fyn ${{ parameters.conda_env }} ${{ parameters.conda_opts }}
 
-      # install reco-utils and needed dependencies
+      # Make sure there is no conda environment
+      conda env remove -n ${{ parameters.conda_env }} || exit -1
+
+      # Create a clean conda environment
+      conda create -fyn ${{ parameters.conda_env }} ${{ parameters.conda_opts }}
       eval "$(conda shell.bash hook)"
       conda activate ${{ parameters.conda_env }}
-      pip install .${{ parameters.pip_opts }}
-      pip install -r dev-requirements.txt
+
+      # install reco-utils and needed dependencies
+      if [ "${{ parameters.install }}" = "release" ]; then
+        echo "   --- SET UP PARAMETERS ---"
+        ARCH=$(uname -m)
+        if [[ $ARCH == "x86_64" ]]; then
+            PLATFORM="manylinux1_x86_64"
+        else
+            PLATFORM="manylinux2014_$ARCH"
+        fi
+        RELEASE_VERSION=$(awk '/__version__ = / {print $3}' reco_utils/__init__.py | sed 's/"//g')
+
+        echo "   --- BUILDING PACKAGE ---"
+        rm -rf dist || exit -1
+
+        python setup.py sdist bdist_wheel --plat-name=$PLATFORM || exit -1
+
+        echo "   --- INSTALLING WHEEL ---"
+        pip install dist/ms_recommenders-$RELEASE_VERSION-py3-none-$PLATFORM.whl${{ parameters.pip_opts }} || exit -1
+      else
+        echo "   --- INSTALLING LATEST CODE ---"
+        pip install .${{ parameters.pip_opts }} || exit -1
+      fi
+      
+      echo "   --- LIST INSTALLED DEPENDENCIES ---"
+      pip list
 
       # clean up any previous test results        
       rm -rf reports
-    displayName: Create Conda Env
-  
+    displayName: 'Install Dependencies'
+
   - ${{ each test in parameters.test_types }}:
     - script: |
         eval "$(conda shell.bash hook)"
+
         conda activate ${{ parameters.conda_env }}
-        export PYSPARK_PYTHON=`which python`
-        export PYSPARK_DRIVER_PYTHON=`which python`
+        pip install pytest>=3.6.4 || exit -1
+
+        if [[ "${{ parameters.conda_env }}" == *"spark"* ]]; then
+          export PYSPARK_PYTHON=`which python`
+          export PYSPARK_DRIVER_PYTHON=`which python`
+          export SPARK_DIR=$PWD/build/spark
+          rm -rf $SPARK_DIR || exit -1
+          mkdir -p $SPARK_DIR || exit -1
+          wget https://mirrors.ukfast.co.uk/sites/ftp.apache.org/spark/spark-2.4.8/spark-2.4.8-bin-hadoop2.7.tgz -O $SPARK_DIR/spark.tgz || exit -1
```

Comment:
i thought installing pip installing pyspark was sufficient, is that not the case? do we need the spark binaries? if that's the case is the version of spark important to synchronize?

Reply:
yeah we need the binaries, in the old test VM we have set up the right SPARK_HOME, but what I wanted to do here is a more portable approach that downloads the binary, sets SPARK_HOME, etc. If in the future, we want to move from DSVM we have now to a hosted machine, it will be just a copy pase. 

About the question of syncronizing the version, that's a good point. Right now we have in setup.py `"pyspark>=2.4.5,<3.0.0"`, that resolves to 2.4.8, which is the same version we have in the binaries. Here I see two options:
1) fix the spark version of the setup.py to 2.4.8
2) get the pyspark version that is installed dynamically (maybe `pip list | grep pyspark | awk "get me the value"` and then input this to the url when we download the binaries

Reply:
Even now, a new VM has newer Spark versions than 2.4.8. So it won't work without downloading the binaries. See [this way of setting up the VM](https://github.com/microsoft/recommenders/issues/1408#issuecomment-848078262).

Reply:
i pretty sure these tests work without requiring binaries, we should test it out before worrying too much about managing the version

Reply:
We have already tested this, they don't work out of the box on a VM.

Reply:
I just spun up a new Ubuntu 18.04 DSVM and it works fine with out installing any binaries.
Here were the steps I took (the key is to unset SPARK_HOME so pip installed spark files are used instead of the pre-existing files on the dsvm):
```
git clone https://github.com/microsoft/recommenders
cd recommenders
conda create -n reco python=3.6 -y
conda activate reco
pip install -e .[spark]
pip install pytest
sudo apt install -y openjdk-8-jdk
export JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk-amd64"
export PYSPARK_PYTHON=`which python`
export PYSPARK_DRIVER_PYTHON=`which python`
unset SPARK_HOME
pytest tests/unit/reco_utils -m "spark and not notebooks and not gpu"
```

Reply:
good catch, I'll review this on another PR

Reply:
You are right, unsetting SPARK_HOME seems to work.

Reply:
PR here: https://github.com/microsoft/recommenders/pull/1444